source/dotnet/Library/AdaptiveCards.Rendering.Wpf/AdaptiveTextBlockRenderer.cs

```diff
@@ -17,6 +17,11 @@ public static class AdaptiveTextBlockRenderer
     {
         public static FrameworkElement Render(AdaptiveTextBlock textBlock, AdaptiveRenderContext context)
         {
+            if (String.IsNullOrEmpty(textBlock.Text))
+            {
+                return null;
+            }
```

Comment:
The problem with this change is that it causes the higher layers of the stack to think that we don't have a TextBlock renderer. This will lead to us triggering fallback, or if there's no fallback returning the warning "No renderer for element TextBlock" which is not correct. I ran into this before which is why i ended up spinning #4305 off of #3702. We need to refactor this code somehow to be able to distinguish between a missing renderer and an element dropped for missing required properties (which i believe we decided should *not* trigger fallback).

Reply:
I believe with the current implementation this change actually works, some code was refactored in order to determine if labels and error messages had to be rendered and as such we look for the first rendereable element with the AdaptiveRenderContext.GetRendereableElement() method.

If we look at the code in line 107, as the rendered ui element is null we're going to skip the rendering of spacing and other properties
https://github.com/microsoft/AdaptiveCards/blob/main/source/dotnet/Library/AdaptiveCards.Rendering.Wpf/AdaptiveContainerRenderer.cs

``` csharp
AdaptiveTypedElement rendereableElement = context.GetRendereableElement(cardElement);

// if there's an element that can be rendered, then render it, otherwise, skip
if (rendereableElement != null)
{
    FrameworkElement uiElement = context.Render(rendereableElement);
    if (uiElement != null)
    { // logic regarding spacing, tags, height and other stuff 
    } 
}

```

Reply:
I haven't tested this PR so I can be wrong

Reply:
sorry, I stand corrected, the warning is being raised

Reply:
![image](https://user-images.githubusercontent.com/35784165/94208317-67430000-fe7e-11ea-82bc-99538290cde4.png)

Reply:
Fix should be done by removing duplicate code from
https://github.com/microsoft/AdaptiveCards/blob/main/source/dotnet/Library/AdaptiveCards.Rendering.Wpf/AdaptiveRenderContext.cs starting from line 245

Seems like there is some left over logic from GetRendereableElement that is duplicated, as we're precalculating the element that can be rendered we could remove the whole if in line 301